### PR TITLE
New Add: Upgrade components in SPFx development with SPSE

### PR DESCRIPTION
**Pull request by WesleyFive**
## Category

- [ ] Content fix
- [x] New article

## Related issues

N/A

## What's in this Pull Request?

Add a new article to introduce how to upgrade components in SPFx development with SPSE.
Signing off by SharePoint Server PG, POC: Wesley Wu

### CODE REVIEW

File: docs/spfx/sharepoint-subscription-edition-component.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+ms.date: 09/08/2023
+ms.localizationpriority: high
+---
+
+
+# Upgrade components in SharePoint Framework development with SharePoint Server Subscription Edition
+
+SharePoint Server Subscription Edition supports SharePoint Framework client-side web parts in classic and modern pages, and extensions in modern pages.
+
+When developing a client-side web part, you can only use the specific versions of components supported by the SharePoint Framework to make the web part compatible with your SharePoint Server.
```
Comment (andrewconnell) [inline docs/spfx/sharepoint-subscription-edition-component.md:13]: Also mention SPFx extensions are those are also supported in SPFx v1.3.x+, such as application customizers & field customizers as they can also use React & Fluent UI React so they'd also be included in this doc, correct?

File: docs/spfx/sharepoint-subscription-edition-component.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+> [!NOTE]
+> Because the component-wide upgrades were not made as part of a broader update to SPFx, some manual configuration of the development environment might be completed in order to utilize them.
+
+> [!IMPORTANT]
+> Only SharePoint Server Subscription Edition (SE) supports the component-wide upgrades for the SPFx development since the [Version 23H2](/sharepoint/what-s-new/new-and-improved-features-in-sharepoint-server-subscription-edition-23h2-release#sharepoint-framework-spfx-component-upgrades) feature update. SharePoint Server 2019 and older SharePoint Server products do not support that.
+
+## Upgrade React and Fluent UI React
+
+The SharePoint Framework v1.4.1 natively supports React v15.8 and Fluent UI React ([formerly Office UI Fabric React](https://devblogs.microsoft.com/microsoft365dev/ui-fabric-is-evolving-into-fluent-ui/)) v5.17.
```
Comment (andrewconnell) [inline docs/spfx/sharepoint-subscription-edition-component.md:25]: This assumes the reader is on SP SE GA... SP SE 23H1 updated SPFx to v1.5.1... should that be incorporated in this statement?

File: docs/spfx/sharepoint-subscription-edition-component.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+If you want to use React v16 or Fluent UI React v7 in your custom web parts developed for SharePoint Server with the SharePoint Framework v1.5.0, here are the instructions:
+
+1. Modify the "dependencies" and "devDependencies" in the npm project:
```
Comment (andrewconnell) [inline docs/spfx/sharepoint-subscription-edition-component.md:29]: All these manual steps to change the versions of SPFx project can be implemented using a script specifying specific versions to install. These steps don't address the **package-lock.json** file which would also need to be updated if you manually edit these version numbers.

But if you instead used a script, it would incorporate those changes in not just the **package.json** but also the **package-lock.json** file. Furthermore, it's generally accepted guidance to use NPM scripts to upgrading NPM packages, not to manually edit the **package.json** file.

For example, the first bullet point could be replaced with:

```console
npm install react@16.13.1 react-dom@16.13.1 @types/react@16.9.51 @types/react-dom@16.9.8 @microsoft/office-ui-fabric-react-bundle@1.15.0 -SE
npm install @microsoft/sp-build-web@1.9.1 @microsoft/sp-module-interfaces@1.9.1 -DE
```

File: docs/spfx/sharepoint-subscription-edition-component.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+    - Change the **react** and **react-dom** versions to "16.13.1", the **@types/react** version to "16.9.51", and the **@types/react-dom** version to "16.9.8".
+
+    - Add **@microsoft/office-ui-fabric-react-bundle**: "1.15.0" in the "dependencies" section.
+
+    - Change the **@microsoft/sp-build-web** and **@microsoft/sp-module-interfaces** versions to "1.9.1" in the "devDependencies" section.
+
+    - Add **gulp-json-editor**: "2.2.1", **@microsoft/sp-tslint-rules**: "1.9.1", and **@microsoft/rush-stack-compiler-2.9**: "0.7.16" to the "devDependencies" section.
+
+    - Change the rest of the SPFx components from version "1.4.0" or "1.4.1" to "1.5.0".
```
Comment (andrewconnell) [inline docs/spfx/sharepoint-subscription-edition-component.md:41]: What about those who deployed SPFx v1.5.1 as part of the 23H1-supported version?

File: docs/spfx/supported-extensibility-platforms-overview.md [hunk 5]
```diff
@@ -20,6 +20,8 @@ Learn more about using SPFx on SharePoint Online: [SPFx support for SharePoint O
 > [!IMPORTANT]
 > SharePoint Server Subscription Edition (SE), announced as a preview in July 2021, supports the same SPFx versions as SharePoint Server 2019.
```
Comment (andrewconnell) [inline docs/spfx/supported-extensibility-platforms-overview.md:21]: Shouldn't this get removed as SPSE 23H1 added support for newer versions than what was supported in SP 2019?

**Review by andrewconnell — CHANGES_REQUESTED** [review]
In addition to the inline comments, what about customers who upgraded/deployed SPFx projects based on v1.5.1 that was added as part of the SP SE 23H1 update?

If 23H2 only includes SPFx v1.5.0, those projects based on SPFx v1.5.1 would break, because the dependencies are no longer there. Seems the docs should at least call that out and warn/advise customers to downgrade their projects.

**Comment by VesaJuvonen** [comment]
Closing based on the chat with the Subscription Edition team - new updates coming for this semester hopefully soon.

**Comment by andrewconnell** [comment]
Would this also apply to #9230?

**Comment by rtonerii** [comment]
So, yeah, this whole SPFx on SPSE thing is feeling like a roadblock, right? We were promised shiny new features and regular updates, but things haven't exactly been at any speed I ever expected. My clients have cold feet (And I'm not talking about the cold weather down here!) about SPSE because of those SPFx limitations, and with SP2019 EOL, there is a large concern here.

Here's the deal, we need Microsoft to get real with us. Are they going to keep pumping up SPFx for SPSE, or is it stuck with not having regular updates forever? SPFx for SPO and on-prem are in no way close, I don't need Graph Services, I could go on but I am trying to keep this short and to the point.

And while we're at it, a little transparency wouldn't hurt. A roadmap, a plan, anything to show us they're not just throwing empty promises into the void. Remember that whole "long-term investment" thing Bill Bär mentioned back in March? Yeah, we're still waiting for the investment part to kick in.

Bottom line, this whole situation stinks, please put some real effort in the ugly stepchild we are all seeing.

**Comment by rtonerii** [comment]
Any update?

The recent SharePoint Server Subscription Edition Version 24H1 release https://learn.microsoft.com/en-us/sharepoint/what-s-new/new-and-improved-features-in-sharepoint-server-subscription-edition-24h1-release is exciting. However, I couldn't find any information about updates to the supported SharePoint Framework (SPFx) version.

Does this imply that SPFx development won't see further improvements within the on-premises Subscription Edition?

Many of my clients require on-premises storage for their content. SPFx integration with Subscription Edition is crucial for their customization needs.

It feels like the combination of SharePoint Server Subscription Edition and SPFx is not receiving the same level of attention (not even close) as SharePoint Online.

Could you please clarify Microsoft's plans for SPFx support within the on-premises Subscription Edition moving forward?
